source/pacmap/pacmap.py

```diff
@@ -847,32 +854,34 @@ class PaCMAP(BaseEstimator):
     '''
 
     def __init__(self,
-                 n_components=2,
-                 n_neighbors=10,
-                 MN_ratio=0.5,
-                 FP_ratio=2.0,
-                 pair_neighbors=None,
-                 pair_MN=None,
-                 pair_FP=None,
-                 distance="euclidean",
-                 lr=1.0,
-                 num_iters=(100, 100, 250),
-                 verbose=False,
-                 apply_pca=True,
-                 intermediate=False,
-                 intermediate_snapshots=[
-                     0, 10, 30, 60, 100, 120, 140, 170, 200, 250, 300, 350, 450],
-                 random_state=None,
-                 save_tree=False
+                 n_components: int = 2,
+                 n_neighbors: Union[int, None] = 10,
+                 MN_ratio: float = 0.5,
+                 FP_ratio: float = 2.0,
+                 pair_neighbors: Optional[np.ndarray] = None,
+                 pair_MN: Optional[np.ndarray] = None,
+                 pair_FP: Optional[np.ndarray] = None,
+                 distance: DistanceMetric = "euclidean",
+                 lr: float = 1.0,
+                 num_iters: Union[Tuple[int, int, int], int] = (100, 100, 250),
+                 verbose: bool = False,
+                 apply_pca: bool = True,
+                 intermediate: bool = False,
+                 intermediate_snapshots: Optional[List[int]] = None,
+                 random_state: Optional[int] = None,
+                 save_tree: bool = False
                  ):
+        if intermediate_snapshots is None:
+            intermediate_snapshots = [0, 10, 30, 60, 100, 120, 140, 170, 200, 250, 300, 350, 450]
+
```

Comment:
I was advised to do this (rather than setting default in function sig) due to how python 3.8 supports types